dart/lib/src/sentry_transaction.dart

```diff
@@ -0,0 +1,24 @@
+import '../sentry.dart';
+
+class SentryTransaction extends SentryEvent {
```

Comment:
What if we make a `EventLike` mixin:

```dart
mixin EventLike<T> {
// basically all fields of an event

  // abstract
  T copyWith();
}
```

Then we could do:
```dart 
class SentryEvent with EventLike<SentryEvent> {
  @override
  SentryEvent copyWith() { ... }
}
```

```dart 
class SentryTransaction with EventLike<SentryTransaction> {
  @override
  SentryTransaction copyWith() { ... }
}
```

That way we could have immutable SentryEvent and SentryTransaction.
The same could also work for ISentrySpan.

Reply:
Well if we want to have immutable transaction and span classes

Reply:
yep, Sentry Java has SentryBaseEvent so all common properties go there since many properties from SentryEvent does not make much sense for SentryTransaction, we'd end up at some point with either abstract class or `mixin`, I'm just trying to make it work the whole setup for now but not changing much, I find it easier to refactor when its already working, but its a good idea